tests/extra_api/server_tcp.c

```diff
@@ -133,15 +134,17 @@ static int _tcp_create_and_bind(struct sockaddr_in *addr)
     int fd;
     int flag;
 
-    fd = socket(PF_INET, SOCK_STREAM, IPPROTO_IP);
+    fd = socket(PF_INET, SOCK_STREAM, IPPROTO_TCP);
     if (!fd) {
         rc = -EBUSY;
         printf("Failed to create socket\n");
         goto err;
     }
 
 #if defined(IB_DEV)
-    rc = setsockopt(fd, SOL_SOCKET, SO_BINDTODEVICE, (void *)IB_DEV, strlen(IB_DEV));
+    struct ifreq ifr = {};
+    strncpy(ifr.ifr_name, IB_DEV, IFNAMSIZ);
+    rc = setsockopt(fd, SOL_SOCKET, SO_BINDTODEVICE, (void *)&ifr, sizeof(ifr));
```

Comment:
according to the man page, the argument is a null-terminated string, Passing ifreq structure is not needed.